README.md

```diff
@@ -14,3 +14,66 @@ source ~/.toolbox/toolbox
 # Add toolbox to your bash_profile to activate it in every SSH session
 echo 'source ~/.toolbox/toolbox' >> ~/.bash_profile
 ```
+
+## How to add new tools
+
+The process for adding new cli tools to the toolbox differs a bit depending on how they are written. If a new tool is written in
+bash, like `sem-install`, `sem-service`, or `retry`, the process is pretty simple:
+ 
+  - bash script will be copied into the toolbox during the `release` step inside the Semaphore CI/CD pipeline. (This happens 
+  inside release/create.sh, function `hosted::create_initial_content`). The release process will be triggered only when a new release is
+  created inside the GitHub repo.
+
+  - In order for the new version of toolbox to be used on Semaphore's hosted agents, toolbox version should be updated [here](https://github.com/renderedtext/s2-platform/blob/master/Makefile). (Needles to say,
+people outside of Semaphore can not perform this step)
+
+  - Later, when the toolbox is being installed on agent machines before a job starts its execution, `install-toolbox` is called.
+  In that script, binaries that have been packaged during the release process mentioned above are just linked, like here:
+
+``` bash
+install_cmd ln -sf ~/.toolbox/sem-context $INSTALL_PATH/sem-context
+install_cmd chmod +x $INSTALL_PATH/sem-context
+if [[ $? -eq 0 ]];then
+  echo "sem-context installed"
+else
+  echo "toolbox_install_error{module='sem-context'} 1" >> /tmp/toolbox_metrics
+fi
+```
+
+### What if your cli tool is written as a separate project in `Go`, for example, like `sem-context` or `cache-cli`
+
+In that case, a few extra steps are neccessary:
+
+  - You must compile those tools within the Semaphore CI/CD pipeline, and upload those binaries to artifacts, so that they would be available when
```

Comment:
@VeljkoMaksimovic 
When I was working on the test results, the binary [got pulled](https://github.com/semaphoreci/toolbox/blob/master/release/create.sh#L152-L156) from the tagged release located in the [repository](https://github.com/semaphoreci/test-results/releases/tag/v0.6.3). We're now including the source in the toolbox repository - `sem-context`, `cache-cli`. 

Correct me if I'm wrong, but new tools should be added to the toolbox like that - and `artifact-cli`, `when`, `spc`, `test-results` releases are treated as a "legacy" method, and we should eventually move them here as well. Correct?

Reply:
The first part is correct, I see that some other cli tools have their separate releases. Is this approach where we put source code for tools inside of the `toolbox` repo a way we want to go forward, and the previous method is legacy: i don't know :D I think @DamjanBecirovic  and @lucaspin are the best people to answer that

Reply:
That is legacy, yes. We intend to transform the toolbox repository into a monorepo, so new CLIs should be added here, like the cache and sem-context CLIs. The old ones will eventually be moved here as well.

Reply:
@lucaspin So that means my explanation here is correct, right? You have a block in the initial pipeline that builds every independent tool (like `sem-context` and `cache-cli` for now) and uploads them to artifacts, which are later pulled during `release` promotion to be accessible when `create.sh` is ran?

Reply:
Yes, that is correct.